tests/media_thumbnail_test.go

```diff
@@ -29,7 +30,10 @@ func TestLocalPngThumbnail(t *testing.T) {
 
 	uri := alice.UploadContent(t, data.LargePng, fileName, contentType)
 
-	fetchAndValidateThumbnail(t, alice, uri)
+	fetchAndValidateThumbnail(t, alice, uri, false)
```

Comment:
There's nothing wrong with this, but for this and the Remote test, I think it'd be better if the old and new endpoints were separate tests.

Supposing you had a defect in one of the versions of endpoints, you'd want the test results to show which version was working fine and which version wasn't.

Equally, if a new cutting-edge homeserver Syndruit wants to skip implementing the legacy endpoints, they may want to blacklist the tests just for the old version.

I asked in the Complement room and it seems doing this as subtests would work for this and it looks fairly easy to do: seems like you just wrap each part of the test like this:

```suggestion
    t.Run("unauthenticated", func(t *testing.T) {
	    fetchAndValidateThumbnail(t, alice, uri, false)
	})
```

& similar for the other ones (The string argument to `t.Run` is a subtest name)